pkg/jaeger/query/trace_query.go

```diff
@@ -262,3 +279,20 @@ func (b *Builder) buildTraceIDSubquery(q *spanstore.TraceQueryParameters) (strin
 	}
 	return query, params
 }
+
+// checkForStatusCodeTags checks the tags map for tags which contain status code values
+// and returns the tag that was found with the value of the status code as represented
+// in the database.
+// This was introduced as part of Jaeger to OTEL conversion logic:
+// https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/a13030ca6208fa7d4477f0805f3b95e271f32b24/pkg/translator/jaeger/jaegerproto_to_traces.go#L218
+func checkForStatusCodeTags(tags map[string]string) (string, string) {
```

Comment:
I am not sure we are doing the right thing here. It seems that otel always convert `tracetranslator.TagError` into span status, so I believe we should check for map if `tracetranslator.TagError` is present and make respective qual.

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/internal/coreinternal/tracetranslator/protospan_translation.go#L21 seem to contain all the translations otel is doing and I believe we should make sure we got it all covered.

Reply:
That particular case is not even checked because its only translated to status code if you use a boolean value but the query parameters are always strings here. 

Not sure if the assumption to treat the string `true` as a boolean value here holds up.

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/a13030ca6208fa7d4477f0805f3b95e271f32b24/pkg/translator/jaeger/jaegerproto_to_traces.go has few of transalator functions as well (eg. `translateHostnameAttr`) I guess we should make sure that is taken into account when querying (maybe it is but I am not fully confident)